src/app/lib/workers/seederTask.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+var _ = require('underscore');
+var fs = require('fs');
+var path = require('path');
+var WebTorrent = require('webtorrent');
+var child = require('child_process');
+var crypt = require('crypto');
+
+var SeederTask = function (opt) {
+    this.webtorrent = null;
+    this.torrentFiles = null;
+    this.tmpLocation = opt.tmpLocation;
+    this.seedLimit = opt.seedLimit;
+    this.connectionLimit = opt.connectionLimit;
+    this.trackerAnnouncement = opt.trackerAnnouncement;
+    this.torrentDir = path.join(this.tmpLocation, 'TorrentCache');
+};
+
+SeederTask.prototype = {
+    start: function() {
+        if (this.webtorrent) {
+            this.stop();
+        }
+
+        process.send('Seeding started');
+
+        var seedTorrentFiles = this.seedLimit ?
+            _.sample(this.getTorrentFiles(), this.seedLimit) : this.getTorrentFiles();
+
+        seedTorrentFiles.forEach(this.joinSwarm.bind(this));
+    },
+
+    stop: function() {
+        if (this.webtorrent) {
+            this.webtorrent.destroy();
+        }
+
+        this.webtorrent = null;
+        this.torrentFiles = [];
+
+        process.send('Seeding stopped');
+        process.kill();
+    },
+
+    getTorrentFiles: function() {
+        if (this.torrentFiles === null) {
+            var regexp = /\.torrent$/i;
+            var files = fs.readdirSync(this.torrentDir);
```

Comment:
instead of calling fs here, i think i'd rather maintain a registery of things that have been 'added/removed' to this module.